moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -511,6 +511,30 @@ void RobotState::setJointGroupPositions(const JointModelGroup* group, const Eige
   updateMimicJoints(group);
 }
 
+void RobotState::setJointGroupActivePositions(const JointModelGroup* group, const std::vector<double>& gstate)
+{
+  assert(gstate.size() == group->getActiveVariableCount());
```

Comment:
...because, here, you test against `getActiveVariableCount()`.

Reply:
then thankfully I got it right at least one time ...

Reply:
Actually this was not right either ... this function only exists in documentation but not in code ... So I could either add it (iterate over `getActiveJointModels` and count `variableSize()` **or** use `getActiveJointModels().size()` and add this to the number of places where multi-variable joints are not correctly supported  **or**  add a member variable that caches the number but that would break ABI compatibility when backporting (but then we don't guarantee that anyway)

Reply:
Hm. I think, I would prefer the last option: adding a member variable to cache that value.

Reply:
done